src/global-messages/GlobalMessage.tsx

```diff
@@ -16,6 +17,8 @@ type Props = {
 
 const GlobalMessage = ({globalMessageContext, style}: Props) => {
   const {language} = useTranslation();
+  const [now, setNow] = useState<number>(Date.now());
+  useInterval(() => setNow(Date.now()), 2500);
```

Comment:
Now there are several places where logic like this is implemented. Even though it is just a couple of lines, I think it is can be a utility function `useNow` under src/utils. It can take the milliseconds as a parameter. So it would just be `const now = useNow(2500)` here.